Models/ProductsListPart.cs

```diff
@@ -49,11 +104,74 @@ public class ProductsListPart : ContentPart {
 
         public ShippingOption ShippingOption {
             get {
-                var fromInfoset = Retrieve<string>("SerializedShippingOption");
-                return string.IsNullOrWhiteSpace(fromInfoset) ? null :
-                    JsonConvert.DeserializeObject<ShippingOption>(fromInfoset);
+                return RetrieveShippingOption();
+            }
+            set {
+                Store(value);
+            }
+        }
+
+        private void Store(ShippingOption shippingOption) {
+            var partElement = GetInfosetElement();
+            var shipElement = GetSubElement(partElement, "ShippingOption");
+            if (shippingOption == null) {
+                shipElement.RemoveAttributes();
+            } else {
+                shipElement.SetAttributeValue("Price", shippingOption.Price);
+                shipElement.SetAttributeValue("Description", shippingOption.Description ?? "");
+                shipElement.SetAttributeValue("ShippingCompany", shippingOption.ShippingCompany ?? "");
+                shipElement.SetAttributeValue("IncludedShippingAreas", string.Join(",",
+                    shippingOption.IncludedShippingAreas.Select(ToStringAndLength)) ?? "");
+                shipElement.SetAttributeValue("ExcludedShippingAreas", string.Join(",",
+                    shippingOption.ExcludedShippingAreas.Select(ToStringAndLength)) ?? "");
+                shipElement.SetAttributeValue("FormValue", shippingOption.FormValue ?? "");
+            }
+        }
+        private string ToStringAndLength(string value) {
+            return string.Format(@"{{{0}}}{{{1}}}", value.Length, value);
+        }
+
+        private ShippingOption RetrieveShippingOption() {
+            var partElement = GetInfosetElement();
+            var shipElement = GetSubElement(partElement, "ShippingOption");
+            return shipElement.Attributes().Any() ? new ShippingOption() {
+                Price = decimal.Parse(shipElement.Attribute("Price").Value),
+                Description = shipElement.Attribute("Description").Value,
+                ShippingCompany = shipElement.Attribute("ShippingCompany").Value,
+                IncludedShippingAreas = FromStringAndLength(shipElement.Attribute("IncludedShippingAreas").Value),
+                ExcludedShippingAreas = FromStringAndLength(shipElement.Attribute("ExcludedShippingAreas").Value),
+                FormValue = shipElement.Attribute("FormValue").Value
+            } : null;
+        }
+        private IEnumerable<string> FromStringAndLength(string serialized) {
```

Comment:
It looks like you came up with your own way to serialize an array of strings. There's a perfectly valid way of doing that in XML:

```xml
<sometag>
  <someOtherTag>value 1</someOtherTag>
  <someOtherTag>value 2</someOtherTag>
</sometag>
```